Client/Frontend/Browser/Tab Management/TabManager.swift

```diff
@@ -601,8 +595,11 @@ class TabManager: NSObject, FeatureFlaggable, TabManagerProtocol {
     func removeTab(_ tab: Tab, completion: (() -> Void)? = nil) {
         guard let index = tabs.firstIndex(where: { $0 === tab }) else { return }
         DispatchQueue.main.async { [unowned self] in
+            // gather the index of the deleted tab within the viable tabs array
+            // so we can select the correct next tab after deletion
+            let viableTabsIndex = deletedIndexForViableTabs(tab)
             self.removeTab(tab, flushToDisk: true)
-            self.updateIndexAfterRemovalOf(tab, deletedIndex: index)
+            self.updateIndexAfterRemovalOf(tab, deletedIndex: index, viableTabsIndex: viableTabsIndex)
```

Comment:
Q. Will this show inactive tabs as part of top tab tray?

Reply:
Remember the discussion about inactive tabs on top tab tray in iPad 🤔

Reply:
With these changes inactive tabs are not shown anymore in top tabs.